README.md

```diff
@@ -239,12 +239,29 @@ The currently used parameters for config are:
 
 `stream_name` The name of the Kinesis stream
 
+`profile` For local use
+
+
+
+KinesisClient can also be configured to push events in batches:
+
+`kinesis_client = NYPLRubyUtil::KinesisClient.new(config, batch_size, automatically_push)`
+
+In this case, the client will wait to push records until at least `batch_size` have been entered. If `automatically_push` is set to false, it will not push the batches until it receives `push_records`
+
+
 #### Usage
 
 `kinesis_client << json_message`
 
 Will encode the `json_message` using the configured avro and write to the configured kinesis stream
 
+If `batch_size` is set, will only write to the kinesis stream when it receives at least `batch_size` many records.
+
+If `automatically_push` is set to false, it will not push records to kinesis until it receives `push_records`
```

Comment:
So when `automatically_push=false`, `batch_size` is effectively ignored, right?

Reply:
When `automatically_push=false`, and `push_records` is called, it still pushes the records in batches of the configured `batch_size`. It just doesn't automatically push records when it reaches the `batch_size`